discourse-doctor

```diff
@@ -0,0 +1,403 @@
+  fi
+  echo "Discourse version at $VERSION_HOSTNAME: $version" | tee -a  $LOG_FILE
+}
+
+##
+## get discourse configuration values from YML file
+##
+
+get_discourse_config() {
+
+  echo -e "\n\n==================== YML SETTINGS ====================" | tee -a $LOG_FILE
+  read_config "DISCOURSE_HOSTNAME"
+  DISCOURSE_HOSTNAME=$read_config_result
+  echo DISCOURSE_HOSTNAME=$DISCOURSE_HOSTNAME  | tee -a  $LOG_FILE
+  local changelog=/tmp/changelog.$PPID
+  read_config "DISCOURSE_SMTP_ADDRESS"
+  SMTP_ADDRESS=$read_config_result
+  echo   SMTP_ADDRESS=$SMTP_ADDRESS | tee -a  $LOG_FILE
+  read_config "DISCOURSE_DEVELOPER_EMAILS"
+  DEVELOPER_EMAILS=$read_config_result
+  echo   DEVELOPER_EMAILS=$DEVELOPER_EMAILS| tee -a  $LOG_FILE
+  read_config "DISCOURSE_SMTP_PASSWORD"
+  SMTP_PASSWORD=$read_config_result
+  echo   SMTP_PASSWORD=$read_config_result
+  echo   SMTP_PASSWORD="(REDACTED)" >> $LOG_FILE
+  read_config "DISCOURSE_SMTP_PORT"
+  SMTP_PORT=$read_config_result
+  echo   SMTP_PORT=$read_config_result| tee -a  $LOG_FILE
+  read_config "DISCOURSE_SMTP_USER_NAME"
+  SMTP_USER_NAME=$read_config_result
+  echo   SMTP_USER_NAME=$read_config_result| tee -a  $LOG_FILE
+  read_config "LETSENCRYPT_ACCOUNT_EMAIL"
+  letsencrypt_account_email=$read_config_result
+  echo "LETSENCRYPT_ACCOUNT_EMAIL=$letsencrypt_account_email"
+  if [ -z "$letsencrypt_account_email" ]
+  then
+    echo "LETSENCRYPT_ACCOUNT_EMAIL is empty" | tee -a $LOG_FILE
+  else
+    echo "LETSENCRYPT_ACCOUNT_EMAIL is set" | tee -a $LOG_FILE
+  fi
+
+}
+
+##
+## get plugins
+##
+get_plugins() {
+  echo -e "\n\n==================== PLUGINS ====================" | tee -a $LOG_FILE
+  grep git containers/$app_name.yml | tee -a $LOG_FILE
+
+  grep git containers/$app_name.yml > /tmp/$PPID.grep
+
+  if grep -cv "github.com/discourse" /tmp/$PPID.grep > /dev/null
+  then
+    echo -e "\nWARNING:" | tee -a $LOG_FILE
+    echo You have what appear to be non-official plugins. | tee -a $LOG_FILE
+    echo "If you are having trouble, you should disable them and try rebuilding again." | tee -a $LOG_FILE
+  fi
+
+}
+
+##
+## output YML file
+##
+dump_yaml() {
+  echo -e "\n\n==================== YML DUMP ====================" | tee -a $LOG_FILE
+  echo Dumping $app_name.yml | tee -a $LOG_FILE
+  sed -e '/\s*#.*$/d' -e '/^\s*$/d' -e 's/  DISCOURSE_SMTP_PASSWORD:\(.*\)/  DISCOURSE_SMTP_PASSWORD: REDACTED/g' containers/$app_name.yml  | tee -a $LOG_FILE
+  echo -e "\n\n" | tee -a $LOG_FILE
+}
+
+##
+## read a variable from the config file
+##
+read_config() {
+  config_line=`egrep "^  #?$1:" $web_file`
+  read_config_result=`echo $config_line | awk  --field-separator=":" '{print $2}'`
+  read_config_result=`echo $read_config_result | sed "s/^\([\"']\)\(.*\)\1\$/\2/g"`
+}
+
+##
+## send a test message
+##
+email_test() {
+  echo -e "\n\n==================== MAIL TEST ===================="
+  echo "I recommend that you get a test address from http://www.mail-tester.com/ , but you "
```

Comment:
Can we phrase it in a more general way please? eg: avoid the "I" :D

Reply:
So noted.

"For a robust test, get an address from..." ?